tools/benchmark/Benchmark.cpp

```diff
@@ -123,8 +126,35 @@ int main(int argc, char** argv) {
             return EXIT_FAILURE;
         }
 
-        const powsybl::iidm::Network& network = powsybl::iidm::Network::readXml(inputStream, powsybl::iidm::converter::ImportOptions(options), powsybl::iidm::converter::FakeAnonymizer());
-        powsybl::iidm::Network::writeXml(outputStream, network, powsybl::iidm::converter::ExportOptions(options));
+        std::unique_ptr<powsybl::iidm::converter::Anonymizer> deanonymizer;
+        const auto& inputMappingFile = vm[MAPPING_FILE].as<std::string>();
+        if (!inputMappingFile.empty()) {
+            deanonymizer = stdcxx::make_unique<powsybl::iidm::converter::SimpleAnonymizer>();
+            std::ifstream inputMappingStream(inputMappingFile);
+            if (!inputMappingStream.is_open()) {
+                std::cerr << stdcxx::format("Unable to open mapping file '%1%' for reading", inputMappingFile) << std::endl;
+                return EXIT_FAILURE;
+            }
+            deanonymizer->read(inputMappingStream);
+            inputMappingStream.close();
+        } else {
+            deanonymizer = stdcxx::make_unique<powsybl::iidm::converter::FakeAnonymizer>();
+        }
+
+        const powsybl::iidm::Network& network = powsybl::iidm::Network::readXml(inputStream, powsybl::iidm::converter::ImportOptions(options), *deanonymizer);
+
+        powsybl::iidm::converter::ExportOptions exportOptions(options);
+        std::unique_ptr<powsybl::iidm::converter::Anonymizer> anonymizer = powsybl::iidm::Network::writeXml(outputStream, network, exportOptions);
+        if (exportOptions.isAnonymized()) {
+            const std::string& outputMappingFile = boost::filesystem::path(outputFile).replace_extension("csv").string();
+            std::ofstream outputMappingStream(outputMappingFile);
+            if (!outputMappingStream.is_open()) {
+                std::cerr << stdcxx::format("Unable to open mapping file '%1%' for writing", outputMappingFile) << std::endl;
+                return EXIT_FAILURE;
+            }
+            anonymizer->write(outputMappingStream);
+            outputMappingStream.close();
+        }
```

Comment:
I think we miss something regarding the Java implementation. The XMLImporter looks for a mapping file near the XIIDM file at the import.
In my mind, the `MAPPING_FILE` was only used for the export, but as the name is based on the output-file, maybe we have done something complicated for nothing.